http.go

```diff
@@ -21,16 +21,11 @@ import (
 // Below is the struct that will implement our ServeHTTP method
 type ServerHandler struct {
```

Comment:
updated `ServerHandler` to simplify and prep for the fx integration by introducing `ServerHandlerMetrics`